source/winUser.py

```diff
@@ -612,3 +616,57 @@ def SendInput(inputs):
 	n = len(inputs)
 	arr = (Input * n)(*inputs)
 	user32.SendInput(n, arr, sizeof(Input))
+
+# Windows Clipboard format constants
+CF_UNICODETEXT = 13
+
+@contextlib.contextmanager
+def openClipboard(hwndOwner=None):
+	"""
+	A context manager version of OpenClipboard from user32. 
+	Use as the expression of a 'with' statement, and CloseClipboard will automatically be called at the end. 
+	"""
+	if not windll.user32.OpenClipboard(hwndOwner):
+		raise ctypes.WinError()
+	try:
+		yield
+	finally:
+		windll.user32.CloseClipboard()
+
+def emptyClipboard():
+	if not windll.user32.EmptyClipboard():
+		raise ctypes.WinError()
+
+def getClipboardData(format):
+	# We only support unicode text for now
+	if format!=CF_UNICODETEXT:
+		raise ValueError("Unsupported format")
+	# Fetch the data from the clipboard as a global memory handle
+	h=windll.user32.GetClipboardData(format)
+	if not h:
+		raise ctypes.WinError()
+	# Lock the global memory  while we fetch the unicode string
+	# But make sure not to free the memory accidentally -- it is not ours
+	h=winKernel.HGLOBAL(h,autoFree=False)
+	with h.lock() as addr:
+		# Read the string from the local memory address
+		return wstring_at(addr)

```

Comment:
I removed the check for addr here as I cannot really see now how it could be 0 as then h would also be 0. but also, if addr were 0, wstring_at would throw WindowsError for NULL.

Reply:
Thanks, this really improves code elegance!